src/components/Pagination/Pagination.tsx

```diff
@@ -2,0 +1,94 @@
+import React, { useEffect } from 'react';
+import cn from 'classnames';
+
+interface Props {
+  total: number;
+  perPage: number;
+  currentPage: number;
+  onPageChange: (e: React.MouseEvent<HTMLAnchorElement>) => void;
+}
+
+const getArrayFromNum = (length: number, startValue: number = 0) => {
+  return Array.from({ length: length }, (_, i) => i + startValue + 1);
+};
+
+export const Pagination: React.FC<Props> = ({
+  total,
+  perPage,
+  currentPage,
+  onPageChange,
+}) => {
+  const pages = getArrayFromNum(Math.ceil(total / perPage));
+  const maxItemsPerPage =
+    pages.length === currentPage ? total % perPage : perPage;
+  const itemsForCurrentPage = getArrayFromNum(
+    maxItemsPerPage,
+    (currentPage - 1) * perPage,
+  );
+
+  useEffect(() => {
+    const params = new URLSearchParams(window.location.search);
+
+    params.set('page', currentPage.toString());
+    params.set('perPage', perPage.toString());
+
+    window.history.replaceState(
+      {},
+      '',
+      `${window.location.pathname}?${params}`,
+    );
+  }, [currentPage, perPage]);
+
+  return (
+    <>
+      <ul className="pagination">
+        <li className={cn('page-item', { disabled: currentPage === 1 })}>
+          <a
+            data-cy="prevLink"
+            className="page-link"
+            href="#prev"
+            aria-disabled={`${currentPage === 1}`}
+            onClick={currentPage === 1 ? undefined : onPageChange}
```

Comment:
Consider calling `e.preventDefault()` in the `onClick` handler to prevent the default anchor behavior, which may cause a page reload.